contrib/epee/include/readline_buffer.h

```diff
@@ -19,12 +21,23 @@ namespace rdln
     }
     void get_line(std::string& line) const;
     void set_prompt(const std::string& prompt);
+    static void add_completion(const std::string& command)
+    {
+      if(std::find(completion_commands.begin(), completion_commands.end(), command) != completion_commands.end())
+	return;
+      completion_commands.push_back(command);
```

Comment:
Ok, this makes more sense. It seemed like readline would have a default history handler.

Now that I expanded the code I see that this is actually duplicating an existing `std::map`. Unfortunate. And `std::sort` + `std::lexicographical_compare` + `std::lower_bound` would be interesting but not particularly noticeable unless one of the apps had some impressive set of commands.